.github/workflows/docker-build.yml

```diff
@@ -0,0 +1,46 @@
+name: Docker
+
+on:
+  workflow_call:
+
+concurrency:
+  group: ${ {github.event_name }}-${{ github.workflow }}-${{ github.ref }}-docker-build
+  cancel-in-progress: ${{github.event_name == 'pull_request'}}
+
+jobs:
+  Build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout kokkos
+        uses: actions/checkout@v3
```

Comment:
We require pinning dependencies.
See
https://github.com/kokkos/kokkos/blob/cf2957859aa45c0666feb65a612cdebdd3c7d539/.github/workflows/clang-format-check.yml#L12
for instance.
That goes for all actions.